packages/core/core/src/Dependency.js

```diff
@@ -33,13 +33,13 @@ type DependencyOpts = {|
 export function createDependency(opts: DependencyOpts): Dependency {
   let id =
     opts.id ||
-    md5FromOrderedObject({
-      sourceAssetId: opts.sourceAssetId,
-      moduleSpecifier: opts.moduleSpecifier,
-      env: opts.env.id,
-      target: opts.target,
-      pipeline: opts.pipeline,
-    });
+    hashString(
+      (opts.sourceAssetId ?? '') +
+        opts.moduleSpecifier +
+        opts.env.id +
+        (opts.target ? JSON.stringify(opts.target) : '') +
+        (opts.pipeline ?? ''),
```

Comment:
target and/or pipeline can be unset here

Reply:
but target will be a JSON object and it's very unlikely that pipeline would look like that